tiers/middleware.py

```diff
@@ -38,12 +39,13 @@ def process_request(self, request):
         except NoReverseMatch:
             pass
 
-        # Nothing to do if the user is not logged in
         if not request.user.is_authenticated:
-            return
+            # Depending on the feature flag, we may redirect the non-logged in users
+            if not should_redirect_non_authenticated():
```

Comment:
using a waffle flag to [decouple release from deployment](https://www.thoughtworks.com/radar/techniques/decoupling-deployment-from-release)